bblocks/import_tools/ilo.py

```diff
@@ -0,0 +1,212 @@
+        glossaries (dict): Dictionary with the names as keys and a dictionary of codes and names as values
+
+    Returns:
+        pd.DataFrame: Cleaned dataframe
+    """
+
+    # loop through glossaries and rename columns
+    for key in glossaries:
+        if key in df.columns:
+            df = df.rename(columns={key: f"{key}_code"})  # rename columns to *_code
+            df[key] = df[f"{key}_code"].map(
+                glossaries[key]
+            )  # add new column with glossary names
+
+    return df
+
+
+def download_data(indicator: str, path: str, glossaries: dict) -> None:
+    """Pipeline to download an indicator and save it to disk.
+
+    Args:
+        indicator: Indicator code to download
+        path: Path to save the data to
+        glossaries: dictionary to map codes to names
+    """
+
+    (extract_data(indicator).pipe(clean_df, glossaries).to_csv(path, index=False))
+
+
+@dataclass
+class ILO(ImportData):
+    """An object to import data from the ILO website
+
+    To use this object, first create an instance of it and then call the `load_data` method
+    to load an indicator to the object. Use the `get_data` method to get the data from the object.
+    To update the data saved on disk and in the object, call the `update_data` method.
+    The attribute `available_indicators` contains a dataframe with information on
+    all the available indicators from the ILO.
+    """
+
+    _glossaries: dict = None  # dictionary used to map codes to names
+    _available_indicators: pd.DataFrame = (
+        None  # dataframe with information on all available indicators
+    )
+
+    @property
+    def available_indicators(self) -> pd.DataFrame:
+        """Return a datafra"""
+
+        if self._available_indicators is None:
+            self._available_indicators = pd.read_csv(
+                f"{BASE_URL}//ilostat-files/WEB_bulk_download/indicator/table_of_contents_en.csv"
+            )
+            logger.info("Loaded available indicators to object")
+        return self._available_indicators
+
+    def _load_glossaries(self) -> None:
+        """Load the glossaries to the object"""
+
+        self._glossaries = get_glossaries()
+        logger.info("Loaded glossaries to object")
+
+    def load_data(self, indicator: str | list) -> ImportData:
+        """Load an ILO indicator to the object
+
+        This method with check if the indicator is already saved to disk. If not, it
+        will downlaod it to disk and load it to the object. Use the `get_data` method
+        to get the data from the object after it is loaded.
+
+        Args:
+            indicator (str | list): Indicator code or list of indicator codes to load
+
+        Returns:
+            ImportData: The object with the data loaded
+        """
+
+        if isinstance(indicator, str):
+            indicator = [indicator]
+
+        for ind in indicator:  # loop through indicators
+
+            path = BBPaths.raw_data / f"{indicator}.csv"
```

Comment:
This is creating a path with the name which includes the entire list. It should be `f"{ind}"`